store/driver/txn/snapshot.go

```diff
@@ -73,6 +73,8 @@ func (s *tikvSnapshot) SetOption(opt int, val interface{}) {
 		s.KVSnapshot.SetNotFillCache(val.(bool))
 	case tikvstore.SnapshotTS:
 		s.KVSnapshot.SetSnapshotTS(val.(uint64))
+	case tikvstore.ReplicaRead:
+		s.KVSnapshot.SetReplicaRead(val.(tikvstore.ReplicaReadType))
```

Comment:
may we meet a bug that the `opt` is ReplicaRead but the `val` is another type?

Reply:
It's true. What I've done so far is just make sure to avoid type conversions in client go. To tweak the whole TiDB code would involve extending the `kv.Transaction` interface, which has too much impact.